adr/0008-named-callbacks.md

```diff
@@ -0,0 +1,55 @@
+# 7: LogAs Function in Capability Class "Named Callbacks"
+
+Date: 2024-03-21
+
+## Status
+
+Proposed
+
+## Context
+
+Pepr utilizes anonymous callbacks that produce logs during execution. This approach, while functional, has led to challenges in debugging and testing. The logs generated by these callbacks lack clear identifiers, making it difficult to ascertain which callback produced which log unless manual names are embedded in the log messages from within the callbacks themselves.
+
+
+## Decision Drivers
+
+The key drivers for this decision include:
+
+- **Devex**: Instead of manual declaration of `const name = "my-action";` in each callback and each log, we can add a name to the callback to make it easier to identify the source of logs. `LogAs("my-action")`
```

Comment:
```suggestion
- **Devex**: Instead of manual declaration of `const name = "my-action";` in each callback and each log (which is easy to forget and tedious to do), we can add a name to the callback to make it easier to identify the source of logs. `LogAs("my-action")`
```